### ISSUE TITLE
API execution not working ('try it out')

### ISSUE THREAD

**Issue by aniket-more48** [SamWSoftware/serverless-auto-swagger#4]
When I execute any API, it does not take the stage context into consideration , hence it fails . 

Eg. - 
Swagger - http://localhost:3000/dev-b/swagger
API : http://localhost:3000/dev-b/test 

When I execute it from swagger (try it out) it runs - http://localhost:3000/test 

Do I need to add any extra configuration(base path) to fix this ?

**Comment by serjant** [comment]
Happens to me too, we need some kind of the basePath configuration

**Comment by morcs** [comment]
I have the same issue on a deployed environment, we're using API mappings with paths (https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-mappings.html) which means our API has a "base path", but the generated swagger references everything at the root /, essentially the same issue.

**Comment by NicoIamarino** [comment]
I made a [pull request](https://github.com/SamWSoftware/serverless-auto-swagger/pull/15) about this issue. Is my first time so I don't know how to link it.

**Comment by bfaulk96** [comment]
As a workaround for you guys until something like this is potentially implemented, you can:
1.  Create some sort of env-specific overrides `swagger-overrides.dev.json` file with the following
    ```json
    {
      "__comment": "To make Swagger calls work in our environments, we need to add basePaths for API mappings",
      "basePath": "/dev-b"
    }
    ```
2. Then in your serverless file (under `custom`), you'd specify something along the lines of
    ```js
    autoswagger: {
      swaggerFiles: ['./swagger-overrides.${self:provider.stage}.json'],
    ```

**Comment by bfaulk96** [comment]
I know it's not super ideal to have a bunch of JSON files just for this – I'm just offering up a temporary fix that I found to be effective 😄

**Comment by bfaulk96** [comment]
`basePath` has been implemented in release `1.7.0`... Will update you when that has been published to `npm` :)

The overall look will be:
```js
custom: {
  ...
  autoswagger: {
    basePath: '/dev-b'
  }
}
```

**Comment by bfaulk96** [comment]
Both `useStage` and `basePath` have been implemented since `1.7.0`. Closing as this has been fixed
